### ISSUE TITLE
conscience agent can not be killed

### ISSUE THREAD

**Issue by jkasarherou** [open-io/oio-sds#1223]
In some cases it appears the conscience agent can not be killed:
we can find this in the logs just before the kill attempt
```
2017-10-11T12:11:30+09:00 node01 OIO,OPENIO,conscienceagent,0: warning  18631 2ABBD41 log WARNING Failed to register service 10.0.0.1:6004: ('Connection aborted.', error(111, 'ECONNREFUSED'))
```

**Comment by jfsmig** [comment]
Related to #1208

**Comment by jkasarherou** [comment]
Could we backport this to 3.X?

**Comment by jfsmig** [comment]
no

**Comment by jfsmig** [comment]
We will backport this in 4.1.x, to let you benefit from the fix ASAP.

**Comment by jkasarherou** [comment]
👍

**Comment by jfsmig** [comment]
Fixed in branch 4.1.x (tag 4.1.5 to come), in branch 4.x (tag 4.2.0 to come) and master.
